Avoid random failures of SaveParticipantTest

This issue is opened on behalf of [Bug 572555](https://bugs.eclipse.org/bugs/show_bug.cgi?id=572555) - Avoid random failures of SaveParticipantTest

Some tests in org.eclipse.jdt.ui.tests.quickfix.SaveParticipantTest fail randomly.

Another failure here: https://download.eclipse.org/eclipse/downloads/drops4/I20260313-0720/reports.html#tests

https://download.eclipse.org/eclipse/downloads/drops4/I20260313-0720/testresults/html/org.eclipse.jdt.ui.tests_ep440I-unit-macosx-x86_64-java21_macosx.cocoa.x86_64_21.html

```
	contents was not formated. Actual: package test1; public class E1 { /** * A Java comment on * two lines */ }

java.lang.AssertionError: contents was not formated. Actual: package test1;
public class E1 {

/**
* A Java comment on
* two lines
*/

}

at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.failEquals(Assert.java:187)
at org.junit.Assert.assertNotEquals(Assert.java:163)
at org.eclipse.jdt.ui.tests.quickfix.SaveParticipantTest.assertChangedFromTo(SaveParticipantTest.java:986)
at org.eclipse.jdt.ui.tests.quickfix.SaveParticipantTest.testBug232768_1(SaveParticipantTest.java:751)
at java.base/java.lang.reflect.Method.invoke(Method.java:580) 
```

Workspace log: https://download.eclipse.org/eclipse/downloads/drops4/I20260313-0720/testresults/ep440I-unit-macosx-x86_64-java21_macosx.cocoa.x86_64_21/org.eclipse.jdt.ui.tests.AutomatedSuite.txt

```

!ENTRY org.eclipse.jdt.ui.tests 1 0 2026-03-13 14:07:38.510
!MESSAGE testIssue2288[{0}] STARTING
| Java indexing | 2026-03-13 14:07:38.528 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.core.JavaCore | getOptions | 4680 | Setting Java options |
| main | 2026-03-13 14:07:38.530 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2859 | JavaModelManager.removeFromSecondaryTypesCache(E1.java) |
| main | 2026-03-13 14:07:38.573 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.574 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypesSearch(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.574 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | 	-> secondary paths stored in cache:  |
| main | 2026-03-13 14:07:38.574 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.581 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.582 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| Worker-3: Delayed Reconciler startup for org.eclipse.jdt.internal.ui.text.JavaReconciler | 2026-03-13 14:07:38.654 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| Worker-3: Delayed Reconciler startup for org.eclipse.jdt.internal.ui.text.JavaReconciler | 2026-03-13 14:07:38.654 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| Worker-3: Delayed Reconciler startup for org.eclipse.jdt.internal.ui.text.JavaReconciler | 2026-03-13 14:07:38.656 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| Worker-3: Delayed Reconciler startup for org.eclipse.jdt.internal.ui.text.JavaReconciler | 2026-03-13 14:07:38.656 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.663 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.663 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.663 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.664 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.682 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.683 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.683 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.683 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.691 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.692 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.692 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.692 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.706 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.707 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.NameLookup | findSecondaryType | 705 | JavaModelManager.secondaryTypes(TestSetupProject,false) |
| main | 2026-03-13 14:07:38.712 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2859 | JavaModelManager.removeFromSecondaryTypesCache(E1.java) |
| main | 2026-03-13 14:07:38.714 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2859 | -> remove file from cache of project: TestSetupProject |
| main | 2026-03-13 14:07:38.718 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2859 | JavaModelManager.removeSecondaryTypesFromMap(,/TestSetupProject/src/test1/E1.java) |
| main | 2026-03-13 14:07:38.719 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2859 | 	- new secondary types map: |
| main | 2026-03-13 14:07:38.721 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2864 | JavaModelManager.removeFromSecondaryTypesCache(E1.java) |
| main | 2026-03-13 14:07:38.721 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2864 | -> remove file from cache of project: TestSetupProject |
| main | 2026-03-13 14:07:38.722 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2864 | JavaModelManager.removeSecondaryTypesFromMap(,/TestSetupProject/src/test1/E1.java) |
| main | 2026-03-13 14:07:38.722 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2864 | 	- new secondary types map: |

!ENTRY org.eclipse.jdt.ui.tests 1 0 2026-03-13 14:07:38.899
!MESSAGE testIssue2288[{0}] DONE: SUCCESSFUL

!ENTRY org.eclipse.jdt.ui.tests 1 0 2026-03-13 14:07:38.899
!MESSAGE testBug232768_1[{0}] STARTING
| Java indexing | 2026-03-13 14:07:38.916 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.core.JavaCore | getOptions | 4680 | Setting Java options |
| main | 2026-03-13 14:07:38.918 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2859 | JavaModelManager.removeFromSecondaryTypesCache(E1.java) |
| main | 2026-03-13 14:07:39.029 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2859 | JavaModelManager.removeFromSecondaryTypesCache(E1.java) |
| main | 2026-03-13 14:07:39.076 | org.eclipse.jdt.core | /debug | org.eclipse.jdt.internal.core.DeltaProcessor | updateIndex | 2864 | JavaModelManager.removeFromSecondaryTypesCache(E1.java) |

!ENTRY org.eclipse.jdt.ui.tests 1 0 2026-03-13 14:07:39.271
!MESSAGE testBug232768_1[{0}] DONE: FAILED
```

--- a/org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/quickfix/CleanUpTestCase.java
+++ b/org.eclipse.jdt.ui.tests/ui/org/eclipse/jdt/ui/tests/quickfix/CleanUpTestCase.java
@@ -29,10 +29,12 @@ import org.junit.Before;
 import org.junit.jupiter.api.Assertions;
 
 import org.eclipse.jdt.testplugin.JavaProjectHelper;
+import org.eclipse.jdt.testplugin.JavaTestPlugin;
 import org.eclipse.jdt.testplugin.TestOptions;
 
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
 import org.eclipse.core.runtime.preferences.IEclipsePreferences;
 import org.eclipse.core.runtime.preferences.InstanceScope;
 
@@ -74,6 +76,7 @@ import org.eclipse.jdt.ui.JavaUI;
 import org.eclipse.jdt.ui.PreferenceConstants;
 import org.eclipse.jdt.ui.cleanup.CleanUpOptions;
 import org.eclipse.jdt.ui.cleanup.ICleanUp;
+import org.eclipse.jdt.ui.tests.util.TestUtils;
 
 import org.eclipse.jdt.internal.ui.JavaPlugin;
 import org.eclipse.jdt.internal.ui.dialogs.StatusInfo;
@@ -144,16 +147,19 @@ public abstract class CleanUpTestCase extends QuickFixTest {
 	}
 
 	protected void disable(String key) throws CoreException {
+		JavaTestPlugin.log(Status.info("Disabling: " + key));
 		fProfile.getSettings().put(key, CleanUpOptions.FALSE);
 		commitProfile();
 	}
 
 	protected void enable(String key) throws CoreException {
+		JavaTestPlugin.log(Status.info("Enabling: " + key));
 		fProfile.getSettings().put(key, CleanUpOptions.TRUE);
 		commitProfile();
 	}
 
 	private void commitProfile() throws CoreException {
+		TestUtils.waitForIndexer();
 		List<Profile> profiles= CleanUpPreferenceUtil.getBuiltInProfiles();
 		profiles.add(fProfile);
 

